Dep init Hangs

### What version of Go (`go version`) and `dep` (`git describe --tags`) are you using?
Go version
`go1.7.4 darwin/amd64`

Dep Version
`v0.1.0-215-g911cd22`


### What `dep` command did you run?
`dep init`

Output:

<!--

Paste the output of the commands you ran in here, making sure to pass -v for maximum context.

The output of `dep hash-inputs` may also be helpful to include.

-->
```
Importing configuration from godep. These are only initial constraints, and are further refined during the solve process.
Detected godep configuration files...
Converting from Godeps.json ...
  Using v0.2.0-9-gf0aeabc as initial hint for imported dep github.com/BurntSushi/toml
  Using ^1.0.0 as initial constraint for imported dep github.com/Jeffail/gabs
  Trying 1.0 (2a3aa15) as initial lock for imported dep github.com/Jeffail/gabs
  Using master as initial constraint for imported dep github.com/RackSec/srslog
  Trying master (a974ba6) as initial lock for imported dep github.com/RackSec/srslog
  Using v0.10.0-16-gcd7d1bb as initial hint for imported dep github.com/Sirupsen/logrus
  Using master as initial constraint for imported dep github.com/armon/consul-api
  Trying master (dcfedd5) as initial lock for imported dep github.com/armon/consul-api
  Using master as initial constraint for imported dep github.com/armon/go-radix
  Trying master (4239b77) as initial lock for imported dep github.com/armon/go-radix
  Using v4-6-gdf81827 as initial hint for imported dep github.com/asaskevich/govalidator
  Using v1.6.27-3-g0876266 as initial hint for imported dep github.com/aws/aws-sdk-go
  Using master as initial constraint for imported dep github.com/bmizerany/perks
  Trying master (d9a9656) as initial lock for imported dep github.com/bmizerany/perks
  Using v2.1.6-1-g5d8c110 as initial hint for imported dep github.com/bsm/sarama-cluster
  Using v2.0.0-38-g003851b as initial hint for imported dep github.com/coreos/go-etcd
  Using ^1.1.0 as initial constraint for imported dep github.com/davecgh/go-spew
  Trying v1.1.0 (346938d) as initial lock for imported dep github.com/davecgh/go-spew
  Using v1.0.0-4-gb86b1ec as initial hint for imported dep github.com/eapache/go-resiliency
  Using master as initial constraint for imported dep github.com/eapache/go-xerial-snappy
  Trying master (bb955e0) as initial lock for imported dep github.com/eapache/go-xerial-snappy
  Using v1.0.2-7-g44cc805 as initial hint for imported dep github.com/eapache/queue
  Using v1.5.0-2-g62e9147 as initial hint for imported dep github.com/fatih/color
  Using v1.4.2-2-gfd9ec7d as initial hint for imported dep github.com/fsnotify/fsnotify
  Using v1.0.0-5-gffa8d46 as initial hint for imported dep github.com/garyburd/redigo
  Using master as initial constraint for imported dep github.com/gin-gonic/contrib
  Trying master (d4fc5a9) as initial lock for imported dep github.com/gin-gonic/contrib
  Using v1.0-2-g3900df0 as initial hint for imported dep github.com/gin-gonic/gin
  Using ^1.25.2 as initial constraint for imported dep github.com/go-ini/ini
  Trying v1.25.2 (74bdc99) as initial lock for imported dep github.com/go-ini/ini
  Using pre-node-events-365-g58b3c71 as initial hint for imported dep github.com/gocql/gocql
  Using master as initial constraint for imported dep github.com/golang/example
  Trying master (0dea2d0) as initial lock for imported dep github.com/golang/example
  Using v1.1-7-g08b5f42 as initial hint for imported dep github.com/gorilla/context
  Using v1.1-10-g801d6e3 as initial hint for imported dep github.com/gorilla/handlers
  Using v1.1-20-g7804150 as initial hint for imported dep github.com/gorilla/mux
  Using v1.1-5-gfa5329f as initial hint for imported dep github.com/gorilla/securecookie
  Using v1.1-2-g83c8db3 as initial hint for imported dep github.com/gorilla/sessions
  Using master as initial constraint for imported dep github.com/hailocab/go-hostpool
  Trying master (e80d13c) as initial lock for imported dep github.com/hailocab/go-hostpool
  Using v0.7.4-140-g5a08348 as initial hint for imported dep github.com/hashicorp/consul
  Using master as initial constraint for imported dep github.com/hashicorp/errwrap
  Trying master (7554cd9) as initial lock for imported dep github.com/hashicorp/errwrap
  Using master as initial constraint for imported dep github.com/hashicorp/go-cleanhttp
  Trying master (3573b8b) as initial lock for imported dep github.com/hashicorp/go-cleanhttp
  Using v0.8.1-26-gd787b2e as initial hint for imported dep github.com/hashicorp/serf
  Using 0.2.2-6-g50d4dbd as initial hint for imported dep github.com/imdario/mergo
  Using master as initial constraint for imported dep github.com/inconshreveable/mousetrap
  Trying master (76626ae) as initial lock for imported dep github.com/inconshreveable/mousetrap
  Using v1.1.0-236-gbcdb0a7 as initial hint for imported dep github.com/influxdata/influxdb
  Using 0.0.5-69-ga643db2 as initial hint for imported dep github.com/jbussdieker/golibxml
  Using 0.2.2-14-gbd40a43 as initial hint for imported dep github.com/jmespath/go-jmespath
  Using master as initial constraint for imported dep github.com/jpillora/backoff
  Trying master (06c7a16) as initial lock for imported dep github.com/jpillora/backoff
  Using v1.1-38-g4563b0b as initial hint for imported dep github.com/julienschmidt/httprouter
  Using 1.0.0-16-g1051eaf as initial hint for imported dep github.com/justinas/alice
  Using ^1.1.0 as initial constraint for imported dep github.com/klauspost/crc32
  Trying v1.1 (cb6bfca) as initial lock for imported dep github.com/klauspost/crc32
  Using master as initial constraint for imported dep github.com/kr/fs
  Trying master (2788f0d) as initial lock for imported dep github.com/kr/fs
  Using go.weekly.2011-12-22-29-gadd1dbc as initial hint for imported dep github.com/kr/pretty
  Using master as initial constraint for imported dep github.com/kr/text
  Trying master (7cafcd8) as initial lock for imported dep github.com/kr/text
  Using go1.0-cutoff-127-gd8eeeb8 as initial hint for imported dep github.com/lib/pq
  Using v1.7.0-5-g0723e35 as initial hint for imported dep github.com/magiconair/properties
  Using master as initial constraint for imported dep github.com/manucorporat/sse
  Trying master (ee05b12) as initial lock for imported dep github.com/manucorporat/sse
  Using v0.0.7-24-gded68f7 as initial hint for imported dep github.com/mattn/go-colorable
  Using ^0.0.2 as initial constraint for imported dep github.com/mattn/go-isatty
  Trying v0.0.2 (fc9e8d8) as initial lock for imported dep github.com/mattn/go-isatty
  Using v0.0.2-3-g97311d9 as initial hint for imported dep github.com/mattn/go-runewidth
  Using v2.0.4-6-geaa1401 as initial hint for imported dep github.com/minio/minio-go
  Using v1.2.0-36-g6fb7c9f as initial hint for imported dep github.com/nats-io/nats
  Using ^1.0.0 as initial constraint for imported dep github.com/nats-io/nuid
  Trying v1.0.0 (289cccf) as initial lock for imported dep github.com/nats-io/nuid
  Using master as initial constraint for imported dep github.com/nu7hatch/gouuid
  Trying master (179d4d0) as initial lock for imported dep github.com/nu7hatch/gouuid
  Using v0.1.0-24-gac55b72 as initial hint for imported dep github.com/oschwald/geoip2-golang
  Using v0.2.0-37-g4b71dbb as initial hint for imported dep github.com/oschwald/maxminddb-golang
  Using v1.0-21-g1b00554 as initial hint for imported dep github.com/pborman/uuid
  Using ^0.1.0 as initial constraint for imported dep github.com/pelletier/go-buffruneio
  Trying v0.1.0 (df1e16f) as initial lock for imported dep github.com/pelletier/go-buffruneio
  Using v0.3.5-16-g45932ad as initial hint for imported dep github.com/pelletier/go-toml
  Using v0.1-4-g5a00444 as initial hint for imported dep github.com/pierrec/xxHash
  Using v0.8.0-2-g248dadf as initial hint for imported dep github.com/pkg/errors
  Using ^1.0.0 as initial constraint for imported dep github.com/pmezard/go-difflib
  Trying v1.0.0 (792786c) as initial lock for imported dep github.com/pmezard/go-difflib
  Using 0.7.0-68-g67994f1 as initial hint for imported dep github.com/prometheus/client_golang
  Using model-0.0.2-12-gfa8ad6f as initial hint for imported dep github.com/prometheus/client_model
  Using master as initial constraint for imported dep github.com/rcrowley/go-metrics
  Trying master (1f30fe9) as initial lock for imported dep github.com/rcrowley/go-metrics
  Using master as initial constraint for imported dep github.com/robmccoll/mitlru
  Trying master (2ba1693) as initial lock for imported dep github.com/robmccoll/mitlru
  Using v1.1.0-6-gb061729 as initial hint for imported dep github.com/satori/go.uuid
  Using v1.1.4-4-g976c720 as initial hint for imported dep github.com/stretchr/testify
  Using master as initial constraint for imported dep github.com/turtlemonvh/altscanner
  Trying master (19d8079) as initial lock for imported dep github.com/turtlemonvh/altscanner
  Using v1.18.0-65-g61f519f as initial hint for imported dep github.com/urfave/cli
  Using master as initial constraint for imported dep github.com/xeipuuv/gojsonpointer
  Trying master (6fe8760) as initial lock for imported dep github.com/xeipuuv/gojsonpointer
  Using master as initial constraint for imported dep github.com/xeipuuv/gojsonreference
  Trying master (e02fc20) as initial lock for imported dep github.com/xeipuuv/gojsonreference
  Using v0.0.2-17-g749e360 as initial hint for imported dep github.com/xordataexchange/crypt
  Using master as initial constraint for imported dep golang.org/x/time
  Trying master (8be79e1) as initial lock for imported dep golang.org/x/time
  Using v3 as initial constraint for imported dep gopkg.in/amz.v3
  Trying v3 (8c3190d) as initial lock for imported dep gopkg.in/amz.v3
  Using ^1.1.0 as initial constraint for imported dep gopkg.in/asn1-ber.v1
  Trying v1.1 (4e86f43) as initial lock for imported dep gopkg.in/asn1-ber.v1
  Using ^8.18.1 as initial constraint for imported dep gopkg.in/go-playground/validator.v8
  Trying v8.18.1 (5f57d22) as initial lock for imported dep gopkg.in/go-playground/validator.v8
  Using 2.0.0-23-g81ebce5 as initial hint for imported dep gopkg.in/gomail.v2
  Using ^0.9.0 as initial constraint for imported dep gopkg.in/inf.v0
  Trying v0.9.0 (3887ee9) as initial lock for imported dep gopkg.in/inf.v0
  Using ^2.5.0 as initial constraint for imported dep gopkg.in/ldap.v2
  Trying v2.5.0 (8168ee0) as initial lock for imported dep gopkg.in/ldap.v2
  Using r2016.08.01-5-g3f83fa5 as initial hint for imported dep gopkg.in/mgo.v2
  Trying v2 (3f83fa5) as initial lock for imported dep gopkg.in/mgo.v2
  Using ^3.0.56 as initial constraint for imported dep gopkg.in/olivere/elastic.v3
  Trying v3.0.56 (1ad75b5) as initial lock for imported dep gopkg.in/olivere/elastic.v3
  Using ^1.2.5 as initial constraint for imported dep gopkg.in/tylerb/graceful.v1
  Trying v1.2.5 (9a3d423) as initial lock for imported dep gopkg.in/tylerb/graceful.v1
  Using ^1.19.1 as initial constraint for imported dep gopkg.in/urfave/cli.v1
  Trying v1.19.1 (0bdedde) as initial lock for imported dep gopkg.in/urfave/cli.v1
  Using ^287.0.0 as initial constraint for imported dep labix.org/v2/mgo
^CSignal received: waiting for 1 ops to complete...
panic: canary - shouldn't be possible this SourceMgr has been released, its methods can no longer be called

goroutine 1 [running]:
panic(0x3fc180, 0xc420202640)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/golang/dep/internal/gps.(*solver).unselectLast(0xc420174a00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:1237 +0x809
github.com/golang/dep/internal/gps.(*solver).backtrack(0xc420174a00, 0xc420348b80)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:987 +0x13e
github.com/golang/dep/internal/gps.(*solver).solve(0xc420174a00, 0x0, 0x0, 0xc4203e1bc0)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:454 +0x791
github.com/golang/dep/internal/gps.(*solver).Solve(0xc420174a00, 0x36, 0x6732c0, 0xc420342030, 0xc4200120d8)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:399 +0xb1
main.(*initCommand).Run(0xc420157bea, 0xc420017b30, 0xc42000c480, 0x0, 0x0, 0x0, 0x0)
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/init.go:171 +0xaff
main.(*Config).Run(0xc420072f60, 0xc420072f60)
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/main.go:158 +0x938
main.main()
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/main.go:45 +0x174
```

### What did you expect to see?
`dep init` would complete.

### What did you see instead?
After printing `Using ^287.0.0 as initial constraint for imported dep labix.org/v2/mgo`, dep init hung for about 20 minutes. When killed, `dep` panicked and printed out a stack trace (pasted above).

Sure! @darkowlzz here's the output:
I let it run and it hung for 5-10 minutes after the line:
`(6)	    try github.com/aws/aws-sdk-go@v1.10.11`
After that, I `ctrl+c`'d it, and it printed a lot more output.

```
Importing configuration from godep. These are only initial constraints, and are further refined during the solve process.
Detected godep configuration files...
  Loading /Users/tchin/Go/src/stash.in.ionicsecurity.com/ip/gyre/Godeps/Godeps.json
Converting from Godeps.json ...
  Using v0.2.0-9-gf0aeabc as initial hint for imported dep github.com/BurntSushi/toml
  Using ^1.0.0 as initial constraint for imported dep github.com/Jeffail/gabs
  Trying 1.0 (2a3aa15) as initial lock for imported dep github.com/Jeffail/gabs
  Using master as initial constraint for imported dep github.com/RackSec/srslog
  Trying master (a974ba6) as initial lock for imported dep github.com/RackSec/srslog
  Using v1.11.0-44-g3efb95d as initial hint for imported dep github.com/Shopify/sarama
  Using v0.10.0-16-gcd7d1bb as initial hint for imported dep github.com/Sirupsen/logrus
  Using master as initial constraint for imported dep github.com/armon/consul-api
  Trying master (dcfedd5) as initial lock for imported dep github.com/armon/consul-api
  Using master as initial constraint for imported dep github.com/armon/go-radix
  Trying master (4239b77) as initial lock for imported dep github.com/armon/go-radix
  Using v4-6-gdf81827 as initial hint for imported dep github.com/asaskevich/govalidator
  Using v1.6.27-3-g0876266 as initial hint for imported dep github.com/aws/aws-sdk-go
  Using master as initial constraint for imported dep github.com/bmizerany/perks
  Trying master (d9a9656) as initial lock for imported dep github.com/bmizerany/perks
  Using v2.1.6-1-g5d8c110 as initial hint for imported dep github.com/bsm/sarama-cluster
  Using v2.0.0-38-g003851b as initial hint for imported dep github.com/coreos/go-etcd
  Using ^1.1.0 as initial constraint for imported dep github.com/davecgh/go-spew
  Trying v1.1.0 (346938d) as initial lock for imported dep github.com/davecgh/go-spew
  Using v1.0.0-4-gb86b1ec as initial hint for imported dep github.com/eapache/go-resiliency
  Using master as initial constraint for imported dep github.com/eapache/go-xerial-snappy
  Trying master (bb955e0) as initial lock for imported dep github.com/eapache/go-xerial-snappy
  Using v1.0.2-7-g44cc805 as initial hint for imported dep github.com/eapache/queue
  Using v1.5.0-2-g62e9147 as initial hint for imported dep github.com/fatih/color
  Using v1.4.2-2-gfd9ec7d as initial hint for imported dep github.com/fsnotify/fsnotify
  Using v1.0.0-5-gffa8d46 as initial hint for imported dep github.com/garyburd/redigo
  Using master as initial constraint for imported dep github.com/gin-gonic/contrib
  Trying master (d4fc5a9) as initial lock for imported dep github.com/gin-gonic/contrib
  Using v1.0-2-g3900df0 as initial hint for imported dep github.com/gin-gonic/gin
  Using ^1.25.2 as initial constraint for imported dep github.com/go-ini/ini
  Trying v1.25.2 (74bdc99) as initial lock for imported dep github.com/go-ini/ini
  Using pre-node-events-365-g58b3c71 as initial hint for imported dep github.com/gocql/gocql
  Using master as initial constraint for imported dep github.com/golang/example
  Trying master (0dea2d0) as initial lock for imported dep github.com/golang/example
  Using v1.1-7-g08b5f42 as initial hint for imported dep github.com/gorilla/context
  Using v1.1-10-g801d6e3 as initial hint for imported dep github.com/gorilla/handlers
  Using v1.1-20-g7804150 as initial hint for imported dep github.com/gorilla/mux
  Using v1.1-5-gfa5329f as initial hint for imported dep github.com/gorilla/securecookie
  Using v1.1-2-g83c8db3 as initial hint for imported dep github.com/gorilla/sessions
  Using master as initial constraint for imported dep github.com/hailocab/go-hostpool
  Trying master (e80d13c) as initial lock for imported dep github.com/hailocab/go-hostpool
  Using v0.7.4-140-g5a08348 as initial hint for imported dep github.com/hashicorp/consul
  Using master as initial constraint for imported dep github.com/hashicorp/errwrap
  Trying master (7554cd9) as initial lock for imported dep github.com/hashicorp/errwrap
  Using master as initial constraint for imported dep github.com/hashicorp/go-cleanhttp
  Trying master (3573b8b) as initial lock for imported dep github.com/hashicorp/go-cleanhttp
  Using v0.8.1-26-gd787b2e as initial hint for imported dep github.com/hashicorp/serf
  Using 0.2.2-6-g50d4dbd as initial hint for imported dep github.com/imdario/mergo
  Using master as initial constraint for imported dep github.com/inconshreveable/mousetrap
  Trying master (76626ae) as initial lock for imported dep github.com/inconshreveable/mousetrap
  Using v1.1.0-236-gbcdb0a7 as initial hint for imported dep github.com/influxdata/influxdb
  Using 0.0.5-69-ga643db2 as initial hint for imported dep github.com/jbussdieker/golibxml
  Using 0.2.2-14-gbd40a43 as initial hint for imported dep github.com/jmespath/go-jmespath
  Using master as initial constraint for imported dep github.com/jpillora/backoff
  Trying master (06c7a16) as initial lock for imported dep github.com/jpillora/backoff
  Using v1.1-38-g4563b0b as initial hint for imported dep github.com/julienschmidt/httprouter
  Using 1.0.0-16-g1051eaf as initial hint for imported dep github.com/justinas/alice
  Using ^1.1.0 as initial constraint for imported dep github.com/klauspost/crc32
  Trying v1.1 (cb6bfca) as initial lock for imported dep github.com/klauspost/crc32
  Using master as initial constraint for imported dep github.com/kr/fs
  Trying master (2788f0d) as initial lock for imported dep github.com/kr/fs
  Using go.weekly.2011-12-22-29-gadd1dbc as initial hint for imported dep github.com/kr/pretty
  Using master as initial constraint for imported dep github.com/kr/text
  Trying master (7cafcd8) as initial lock for imported dep github.com/kr/text
  Using go1.0-cutoff-127-gd8eeeb8 as initial hint for imported dep github.com/lib/pq
  Using v1.7.0-5-g0723e35 as initial hint for imported dep github.com/magiconair/properties
  Using master as initial constraint for imported dep github.com/manucorporat/sse
  Trying master (ee05b12) as initial lock for imported dep github.com/manucorporat/sse
  Using v0.0.7-24-gded68f7 as initial hint for imported dep github.com/mattn/go-colorable
  Using ^0.0.2 as initial constraint for imported dep github.com/mattn/go-isatty
  Trying v0.0.2 (fc9e8d8) as initial lock for imported dep github.com/mattn/go-isatty
  Using v0.0.2-3-g97311d9 as initial hint for imported dep github.com/mattn/go-runewidth
  Using v2.0.4-6-geaa1401 as initial hint for imported dep github.com/minio/minio-go
  Using v1.2.0-36-g6fb7c9f as initial hint for imported dep github.com/nats-io/nats
  Using ^1.0.0 as initial constraint for imported dep github.com/nats-io/nuid
  Trying v1.0.0 (289cccf) as initial lock for imported dep github.com/nats-io/nuid
  Using master as initial constraint for imported dep github.com/nu7hatch/gouuid
  Trying master (179d4d0) as initial lock for imported dep github.com/nu7hatch/gouuid
  Using v0.1.0-24-gac55b72 as initial hint for imported dep github.com/oschwald/geoip2-golang
  Using v0.2.0-37-g4b71dbb as initial hint for imported dep github.com/oschwald/maxminddb-golang
  Using v1.0-21-g1b00554 as initial hint for imported dep github.com/pborman/uuid
  Using ^0.1.0 as initial constraint for imported dep github.com/pelletier/go-buffruneio
  Trying v0.1.0 (df1e16f) as initial lock for imported dep github.com/pelletier/go-buffruneio
  Using v0.3.5-16-g45932ad as initial hint for imported dep github.com/pelletier/go-toml
  Using v0.1-4-g5a00444 as initial hint for imported dep github.com/pierrec/xxHash
  Using v0.8.0-2-g248dadf as initial hint for imported dep github.com/pkg/errors
  Using ^1.0.0 as initial constraint for imported dep github.com/pmezard/go-difflib
  Trying v1.0.0 (792786c) as initial lock for imported dep github.com/pmezard/go-difflib
  Using 0.7.0-68-g67994f1 as initial hint for imported dep github.com/prometheus/client_golang
  Using model-0.0.2-12-gfa8ad6f as initial hint for imported dep github.com/prometheus/client_model
  Using master as initial constraint for imported dep github.com/rcrowley/go-metrics
  Trying master (1f30fe9) as initial lock for imported dep github.com/rcrowley/go-metrics
  Using master as initial constraint for imported dep github.com/robmccoll/mitlru
  Trying master (2ba1693) as initial lock for imported dep github.com/robmccoll/mitlru
  Using v1.1.0-6-gb061729 as initial hint for imported dep github.com/satori/go.uuid
  Using v1.1.4-4-g976c720 as initial hint for imported dep github.com/stretchr/testify
  Using master as initial constraint for imported dep github.com/turtlemonvh/altscanner
  Trying master (19d8079) as initial lock for imported dep github.com/turtlemonvh/altscanner
  Using v1.18.0-65-g61f519f as initial hint for imported dep github.com/urfave/cli
  Using master as initial constraint for imported dep github.com/xeipuuv/gojsonpointer
  Trying master (6fe8760) as initial lock for imported dep github.com/xeipuuv/gojsonpointer
  Using master as initial constraint for imported dep github.com/xeipuuv/gojsonreference
  Trying master (e02fc20) as initial lock for imported dep github.com/xeipuuv/gojsonreference
  Using v0.0.2-17-g749e360 as initial hint for imported dep github.com/xordataexchange/crypt
  Using master as initial constraint for imported dep golang.org/x/time
  Trying master (8be79e1) as initial lock for imported dep golang.org/x/time
  Using v3 as initial constraint for imported dep gopkg.in/amz.v3
  Trying v3 (8c3190d) as initial lock for imported dep gopkg.in/amz.v3
  Using ^1.1.0 as initial constraint for imported dep gopkg.in/asn1-ber.v1
  Trying v1.1 (4e86f43) as initial lock for imported dep gopkg.in/asn1-ber.v1
  Using ^8.18.1 as initial constraint for imported dep gopkg.in/go-playground/validator.v8
  Trying v8.18.1 (5f57d22) as initial lock for imported dep gopkg.in/go-playground/validator.v8
  Using 2.0.0-23-g81ebce5 as initial hint for imported dep gopkg.in/gomail.v2
  Using ^0.9.0 as initial constraint for imported dep gopkg.in/inf.v0
  Trying v0.9.0 (3887ee9) as initial lock for imported dep gopkg.in/inf.v0
  Using ^2.5.0 as initial constraint for imported dep gopkg.in/ldap.v2
  Trying v2.5.0 (8168ee0) as initial lock for imported dep gopkg.in/ldap.v2
  Using r2016.08.01-5-g3f83fa5 as initial hint for imported dep gopkg.in/mgo.v2
  Trying v2 (3f83fa5) as initial lock for imported dep gopkg.in/mgo.v2
  Using ^3.0.56 as initial constraint for imported dep gopkg.in/olivere/elastic.v3
  Trying v3.0.56 (1ad75b5) as initial lock for imported dep gopkg.in/olivere/elastic.v3
  Using ^1.2.5 as initial constraint for imported dep gopkg.in/tylerb/graceful.v1
  Trying v1.2.5 (9a3d423) as initial lock for imported dep gopkg.in/tylerb/graceful.v1
  Using ^1.19.1 as initial constraint for imported dep gopkg.in/urfave/cli.v1
  Trying v1.19.1 (0bdedde) as initial lock for imported dep gopkg.in/urfave/cli.v1
  Using ^287.0.0 as initial constraint for imported dep labix.org/v2/mgo
Root project is "stash.in.ionicsecurity.com/ip/gyre"
 295 transitively valid internal packages
 89 external packages imported from 79 projects
(0)   ✓ select (root)
(1)	? attempt github.com/BurntSushi/toml with 1 pkgs; 8 versions to try
(1)	    try github.com/BurntSushi/toml@v0.2.0-9-gf0aeabc
(1)	✓ select github.com/BurntSushi/toml@v0.2.0-9-gf0aeabc w/1 pkgs
(2)	? attempt github.com/Jeffail/gabs with 1 pkgs; at least 1 versions to try
(2)	    try github.com/Jeffail/gabs@1.0
(2)	✓ select github.com/Jeffail/gabs@1.0 w/1 pkgs
(3)	? attempt github.com/armon/go-radix with 1 pkgs; at least 1 versions to try
(3)	    try github.com/armon/go-radix@master
(3)	✓ select github.com/armon/go-radix@master w/1 pkgs
(4)	? attempt github.com/RackSec/srslog with 1 pkgs; at least 1 versions to try
(4)	    try github.com/RackSec/srslog@master
(4)	✓ select github.com/RackSec/srslog@master w/1 pkgs
(5)	? attempt github.com/fsnotify/fsnotify with 1 pkgs; 33 versions to try
(5)	    try github.com/fsnotify/fsnotify@v1.4.2-2-gfd9ec7d
(5)	✓ select github.com/fsnotify/fsnotify@v1.4.2-2-gfd9ec7d w/1 pkgs
(6)	? attempt github.com/aws/aws-sdk-go with 5 pkgs; at least 1 versions to try
(6)	    try github.com/aws/aws-sdk-go@08762661fb8f2c081b434de78a363daf6e8615a0
(6)	✗   unable to update repository:
(6)	    try github.com/aws/aws-sdk-go@v1.10.11
^CSignal received: waiting for 1 ops to complete...
(6)	✗   context canceled
(6)	    try github.com/aws/aws-sdk-go@v1.10.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.10.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.9.44
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.9.00
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.9.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.44
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.43
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.42
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.41
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.40
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.39
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.38
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.37
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.36
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.35
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.34
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.33
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.32
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.31
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.30
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.29
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.28
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.27
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.26
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.25
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.24
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.23
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.22
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.21
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.20
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.19
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.18
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.17
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.16
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.15
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.14
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.8.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.7.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.27
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.26
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.25
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.24
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.23
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.22
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.21
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.20
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.19
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.18
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.17
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.16
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.15
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.14
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.6.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.5.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.22
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.21
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.20
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.19
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.18
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.17
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.16
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.15
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@1.4.14
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@1.4.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.4.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.3.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.3.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.2.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.36
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.35
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.34
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.33
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.32
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.31
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.30
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.29
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.28
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.27
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.26
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.25
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.24
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.23
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.22
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.21
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.20
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.19
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.18
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.17
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.16
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.15
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.14
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.1.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v1.0.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.10.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.10.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.10.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.10.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.10.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.17
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.16
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.15
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.14
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.13
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.12
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.11
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.10
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.7.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.7.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.7.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.7.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.7.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.8
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.7
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.6
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.6.0
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@master
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@release
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v.1.2.9
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.0rc1
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.1rc2
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.2rc3
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.3rc4
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	    try github.com/aws/aws-sdk-go@v0.9.4rc5
(6)	✗   this SourceMgr has been released, its methods can no longer be called
(6)	  ← no more versions of github.com/aws/aws-sdk-go to try; begin backtrack
panic: canary - shouldn't be possible this SourceMgr has been released, its methods can no longer be called

goroutine 1 [running]:
panic(0x3fc180, 0xc4202b30a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/golang/dep/internal/gps.(*solver).unselectLast(0xc420195800, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:1237 +0x809
github.com/golang/dep/internal/gps.(*solver).backtrack(0xc420195800, 0xc42033dea0)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:987 +0x13e
github.com/golang/dep/internal/gps.(*solver).solve(0xc420195800, 0x0, 0x0, 0xc420147050)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:454 +0x791
github.com/golang/dep/internal/gps.(*solver).Solve(0xc420195800, 0x36, 0x6732c0, 0xc4201ec030, 0xc4200140d8)
	/Users/tchin/Go/src/github.com/golang/dep/internal/gps/solver.go:399 +0xb1
main.(*initCommand).Run(0xc420171aaa, 0xc4200b39f0, 0xc4200781d0, 0x0, 0x0, 0x0, 0x0)
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/init.go:171 +0xaff
main.(*Config).Run(0xc420072f60, 0xc420072f60)
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/main.go:158 +0x938
main.main()
	/Users/tchin/Go/src/github.com/golang/dep/cmd/dep/main.go:45 +0x174
```

--- a/internal/gps/cmd.go
+++ b/internal/gps/cmd.go
@@ -45,14 +45,14 @@ func (c *monitoredCmd) run(ctx context.Context) error {
 		return ctx.Err()
 	}
 
-	ticker := time.NewTicker(c.timeout)
-	done := make(chan error, 1)
-	defer ticker.Stop()
-
 	err := c.cmd.Start()
 	if err != nil {
 		return err
 	}
+
+	ticker := time.NewTicker(c.timeout)
+	defer ticker.Stop()
+	done := make(chan error, 1)
 
 	go func() {
 		done <- c.cmd.Wait()
@@ -62,14 +62,14 @@ func (c *monitoredCmd) run(ctx context.Context) error {
 		select {
 		case <-ticker.C:
 			if c.hasTimedOut() {
-				if err := c.cmd.Process.Kill(); err != nil {
+				if err := killProcess(c.cmd); err != nil {
 					return &killCmdError{err}
 				}
 
 				return &timeoutError{c.timeout}
 			}
 		case <-ctx.Done():
-			if err := c.cmd.Process.Kill(); err != nil {
+			if err := killProcess(c.cmd); err != nil {
 				return &killCmdError{err}
 			}
 			return ctx.Err()
@@ -90,6 +90,7 @@ func (c *monitoredCmd) combinedOutput(ctx context.Context) ([]byte, error) {
 		return c.stderr.buf.Bytes(), err
 	}
 
+	// FIXME(sdboyer) this is not actually combined output
 	return c.stdout.buf.Bytes(), nil
 }
 

--- a/internal/gps/cmd.go
+++ b/internal/gps/cmd.go
@@ -10,6 +10,7 @@ import (
 	"fmt"
 	"os/exec"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/Masterminds/vcs"
@@ -25,6 +26,18 @@ type monitoredCmd struct {
 	stderr  *activityBuffer
 }
 
+// timeoutError indicates that the monitored process was terminated due to
+// exceeding activity timeouts.
+type timeoutError struct {
+	timeout time.Duration
+}
+
+// killCmdError indicates that an error occurred while sending a kill signal to
+// the monitored process.
+type killCmdError struct {
+	err error
+}
+
 func newMonitoredCmd(cmd *exec.Cmd, timeout time.Duration) *monitoredCmd {
 	stdout, stderr := newActivityBuffer(), newActivityBuffer()
 	cmd.Stdout, cmd.Stderr = stdout, stderr
@@ -37,8 +50,8 @@ func newMonitoredCmd(cmd *exec.Cmd, timeout time.Duration) *monitoredCmd {
 }
 
 // run will wait for the command to finish and return the error, if any. If the
-// command does not show any activity for more than the specified timeout the
-// process will be killed.
+// command does not show any progress, as indicated by writing to stdout or
+// stderr, for more than the specified timeout, the process will be killed.
 func (c *monitoredCmd) run(ctx context.Context) error {
 	// Check for cancellation before even starting
 	if ctx.Err() != nil {
@@ -52,31 +65,58 @@ func (c *monitoredCmd) run(ctx context.Context) error {
 
 	ticker := time.NewTicker(c.timeout)
 	defer ticker.Stop()
+
+	// Atomic marker to track proc exit state. Guards against bad channel
+	// select receive order, where a tick or context cancellation could come
+	// in at the same time as process completion, but one of the former are
+	// picked first; in such a case, cmd.Process could(?) be nil by the time we
+	// call signal methods on it.
+	var isDone *int32 = new(int32)
 	done := make(chan error, 1)
 
 	go func() {
+		// Wait() can only be called once, so this must act as the completion
+		// indicator for both normal *and* signal-induced termination.
 		done <- c.cmd.Wait()
+		atomic.CompareAndSwapInt32(isDone, 0, 1)
 	}()
 
+	var killerr error
+selloop:
 	for {
 		select {
-		case <-ticker.C:
-			if c.hasTimedOut() {
-				if err := killProcess(c.cmd); err != nil {
-					return &killCmdError{err}
-				}
-
-				return &timeoutError{c.timeout}
-			}
-		case <-ctx.Done():
-			if err := killProcess(c.cmd); err != nil {
-				return &killCmdError{err}
-			}
-			return ctx.Err()
 		case err := <-done:
 			return err
+		case <-ticker.C:
+			if !atomic.CompareAndSwapInt32(isDone, 1, 1) && c.hasTimedOut() {
+				if err := killProcess(c.cmd, isDone); err != nil {
+					killerr = &killCmdError{err}
+				} else {
+					killerr = &timeoutError{c.timeout}
+				}
+				break selloop
+			}
+		case <-ctx.Done():
+			if !atomic.CompareAndSwapInt32(isDone, 1, 1) {
+				if err := killProcess(c.cmd, isDone); err != nil {
+					killerr = &killCmdError{err}
+				} else {
+					killerr = ctx.Err()
+				}
+				break selloop
+			}
 		}
 	}
+
+	// This is only reachable on the signal-induced termination path, so block
+	// until a message comes through the channel indicating that the command has
+	// exited.
+	//
+	// TODO(sdboyer) if the signaling process errored (resulting in a
+	// killCmdError stored in killerr), is it possible that this receive could
+	// block forever on some kind of hung process?
+	<-done
+	return killerr
 }
 
 func (c *monitoredCmd) hasTimedOut() bool {
@@ -121,16 +161,8 @@ func (b *activityBuffer) lastActivity() time.Time {
 	return b.lastActivityStamp
 }
 
-type timeoutError struct {
-	timeout time.Duration
-}
-
 func (e timeoutError) Error() string {
 	return fmt.Sprintf("command killed after %s of no activity", e.timeout)
-}
-
-type killCmdError struct {
-	err error
 }
 
 func (e killCmdError) Error() string {

--- a/internal/gps/cmd.go
+++ b/internal/gps/cmd.go
@@ -26,9 +26,9 @@ type monitoredCmd struct {
 	stderr  *activityBuffer
 }
 
-// timeoutError indicates that the monitored process was terminated due to
-// exceeding activity timeouts.
-type timeoutError struct {
+// noProgressError indicates that the monitored process was terminated due to
+// exceeding exceeding the progress timeout.
+type noProgressError struct {
 	timeout time.Duration
 }
 
@@ -92,7 +92,7 @@ selloop:
 				if err := killProcess(c.cmd, isDone); err != nil {
 					killerr = &killCmdError{err}
 				} else {
-					killerr = &timeoutError{c.timeout}
+					killerr = &noProgressError{c.timeout}
 				}
 				break selloop
 			}
@@ -161,7 +161,7 @@ func (b *activityBuffer) lastActivity() time.Time {
 	return b.lastActivityStamp
 }
 
-func (e timeoutError) Error() string {
+func (e noProgressError) Error() string {
 	return fmt.Sprintf("command killed after %s of no activity", e.timeout)
 }
 

--- a/internal/gps/cmd_test.go
+++ b/internal/gps/cmd_test.go
@@ -16,7 +16,7 @@ import (
 func mkTestCmd(iterations int) *monitoredCmd {
 	return newMonitoredCmd(
 		exec.Command("./echosleep", "-n", fmt.Sprint(iterations)),
-		500*time.Millisecond,
+		490*time.Millisecond,
 	)
 }
 
@@ -49,7 +49,7 @@ func TestMonitoredCmd(t *testing.T) {
 		t.Error("Expected command to fail")
 	}
 
-	_, ok := err.(*timeoutError)
+	_, ok := err.(*noProgressError)
 	if !ok {
 		t.Errorf("Expected a timeout error, but got: %s", err)
 	}

--- a/internal/gps/cmd_unix.go
+++ b/internal/gps/cmd_unix.go
@@ -1,0 +1,53 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !windows
+
+package gps
+
+import (
+	"os"
+	"os/exec"
+	"time"
+)
+
+// killProcess manages the termination of subprocesses in a way that tries to be
+// gentle (via os.Interrupt), but resorts to Kill if needed.
+//
+//
+// TODO(sdboyer) should the return differentiate between whether gentle methods
+// succeeded vs. falling back to a hard kill?
+func killProcess(cmd *exec.Cmd) error {
+	if err := cmd.Process.Signal(os.Interrupt); err != nil {
+		// If an error comes back from attempting to signal, proceed immediately
+		// to hard kill.
+		return cmd.Process.Kill()
+	}
+
+	// If the process doesn't exit immediately, check every 50ms, up to 3s,
+	// after which send a hard kill.
+	//
+	// Cannot rely on cmd.ProcessState.Exited() here, as that is not set
+	// correctly when the process exits due to a signal. See
+	// https://github.com/golang/go/issues/19798
+	if cmd.ProcessState == nil || !cmd.ProcessState.Exited() {
+		to := time.NewTimer(3 * time.Second)
+		tick := time.NewTicker(50 * time.Millisecond)
+
+		defer to.Stop()
+		defer tick.Stop()
+
+		// Loop until the ProcessState shows up, indicating the proc has exited,
+		// or the timer expires and
+		for cmd.ProcessState != nil {
+			select {
+			case <-to.C:
+				return cmd.Process.Kill()
+			case <-tick.C:
+			}
+		}
+	}
+
+	return nil
+}

--- a/internal/gps/cmd_unix.go
+++ b/internal/gps/cmd_unix.go
@@ -9,6 +9,7 @@ package gps
 import (
 	"os"
 	"os/exec"
+	"sync/atomic"
 	"time"
 )
 
@@ -18,7 +19,7 @@ import (
 //
 // TODO(sdboyer) should the return differentiate between whether gentle methods
 // succeeded vs. falling back to a hard kill?
-func killProcess(cmd *exec.Cmd) error {
+func killProcess(cmd *exec.Cmd, isDone *int32) error {
 	if err := cmd.Process.Signal(os.Interrupt); err != nil {
 		// If an error comes back from attempting to signal, proceed immediately
 		// to hard kill.
@@ -30,8 +31,10 @@ func killProcess(cmd *exec.Cmd) error {
 	//
 	// Cannot rely on cmd.ProcessState.Exited() here, as that is not set
 	// correctly when the process exits due to a signal. See
-	// https://github.com/golang/go/issues/19798
-	if cmd.ProcessState == nil || !cmd.ProcessState.Exited() {
+	// https://github.com/golang/go/issues/19798 . Also cannot rely on it
+	// because cmd.ProcessState will be nil before the process exits, and
+	// checking if nil create a data race.
+	if !atomic.CompareAndSwapInt32(isDone, 1, 1) {
 		to := time.NewTimer(3 * time.Second)
 		tick := time.NewTicker(50 * time.Millisecond)
 
@@ -40,7 +43,7 @@ func killProcess(cmd *exec.Cmd) error {
 
 		// Loop until the ProcessState shows up, indicating the proc has exited,
 		// or the timer expires and
-		for cmd.ProcessState != nil {
+		for !atomic.CompareAndSwapInt32(isDone, 1, 1) {
 			select {
 			case <-to.C:
 				return cmd.Process.Kill()

--- a/internal/gps/cmd_windows.go
+++ b/internal/gps/cmd_windows.go
@@ -1,0 +1,14 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build windows
+
+package gps
+
+import "os/exec"
+
+func killProcess(cmd *exec.Cmd) error {
+	// TODO it'd be great if this could be more sophisticated...
+	return cmd.Process.Kill()
+}

--- a/internal/gps/cmd_windows.go
+++ b/internal/gps/cmd_windows.go
@@ -8,7 +8,7 @@ package gps
 
 import "os/exec"
 
-func killProcess(cmd *exec.Cmd) error {
+func killProcess(cmd *exec.Cmd, isDone *int32) error {
 	// TODO it'd be great if this could be more sophisticated...
 	return cmd.Process.Kill()
 }

